### ISSUE TITLE
Download updated firmware?

### ISSUE THREAD

**Issue by odlg** [argrento/huami-token#12]
Would it be possible to extend this to download new firmware for Huami devices?

**Comment by argrento** [comment]
Yes, I think it would be possible. At least I can not see any problems with implementing this feature.

**Comment by TimelessNL** [comment]
This would be a really nice feature!

I recently acquired a Mi Band 5, and I was wondering how to update the firmware in the future (_while running GadgetBridge_).

Since GadgetBridge is unable to distribute the firmware, the only official way to update the device is to (_re_)install the Mi Fit app and (_re_)pair the device. But this will probably also reset the auth-key?

The unofficial way would be to download them from this [topic](https://geekdoing.com/threads/mi-band-5-original-firmwares-resources-fonts.1952/)  that distributes (_most likely official_) firmwares for the MB5 using [mega.nz](https://mega.nz/folder/RtNGBICI#Tum-PKqpZk9RKznjZmEVYQ),  but I personally would like to download the firmware's myself to ensure they are authentic.

Having huami-token to acquire the firmware/res/font files would be a nice addition 👍.

**Comment by TimelessNL** [comment]
@argrento, great that this feature is implemented 🥳 .
**Very much appreciated.**

Unfortunately there are 2 issues for me:
1. The string->int cast [here](https://github.com/argrento/huami-token/blob/24033ac2c15fb07a0d83a1b745b3d2f6a56dddaa/huami_token.py#L353) seems incorrect, which is causing:
```
ID of the device to check for updates (-1 for all of them): 0
Traceback (most recent call last):
  File "./huami_token.py", line 354, in <module>
    device.get_firmware(wearables[wearable_id])
TypeError: list indices must be integers or slices, not str
```
2. What device did you use during development, my MiBand5 (_it being on V1.0.1.84_) clearly needs an update but unfortunately the script seems to suggest otherwise (_I printed some of the variables to see what is being send/received_):
```
ID of the device to check for updates (-1 for all of them): -1
Be extremely careful with downloaded files!

Checking for device 0...
{'productionSource': '257', 'deviceSource': '59', 'fontVersion': '0', 'fontFlag': '0', 'appVersion': '5.9.2-play_100355', 'firmwareVersion': 'V1.0.1.84', 'hardwareVersion': 'V0.44.18.2', 'support8Bytes': 'true'}
<Response [200]>
{'deviceType': 'ALL', 'deviceSource': 59, 'ignore': False}
No updates found!
```
The `5.9.2-play_100355` number seems to suggest the [Zepp ](https://www.apkmirror.com/apk/huami/amazfit-watch/amazfit-watch-5-9-2-play-release/zepp-formerly-amazfit-5-9-2-play-android-apk-download/) app and not [Mi Fit](https://www.apkmirror.com/apk/anhui-huami-information-technology-co-ltd/mi-fit/). Or Am I incorrectly assuming this should work on the Mi Bands?

**Comment by argrento** [comment]
1. Forgot to test that case. Fixed it.
2. I'll do some more tests with my Mi Band 5.

Zepp can work correctly with Mi Band (checked by myself). I'll try to figure out with my Mi Band 5, what's the problem. 

Thanks!
